File: Week 1/Cash or Credit/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long n = get_long("Number: ");
    long temp = n, sum = 0, len = 0;

    while (temp != 0)
    {
        temp /= 10;
        sum += (2 * (temp % 10)) % 10 + ((2 * (temp % 10)) / 10) % 10;
        temp /= 10;
    }

    temp = n;
    while (temp != 0)
    {
        sum += temp % 10;
        temp /= 100;
    }

    temp = n;
    while (temp != 0)
    {
        len++;
        temp /= 10;
    }

    if (sum % 10 != 0)
    {
        printf("INVALID\n");
    }
    else if (len == 15 && (n / 10000000000000 == 34 || n / 10000000000000 == 37))
    {
        printf("AMEX\n");
    }
    else if (len == 16 && (n / 100000000000000 > 50 && n / 100000000000000 < 56))
    {
        printf("MASTERCARD\n");
    }
    else if ((len == 13 && n / 1000000000000 == 4) || (len == 16 && n / 1000000000000000 == 4))
    {
        printf("VISA\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

File: Week 6/Sentimental/Cash or Credit/credit.py
import re
from cs50 import get_string

n = get_string("Number: ")
ans = ""

if re.match("^3[47][0-9]{13}$", n):
    ans = "AMEX\n"
elif re.match("^5[12345][0-9]{14}$", n):
    ans = "MASTERCARD\n"
elif re.match("^4[0-9]{12,15}$", n):
    ans = "VISA\n"
else:
    ans = "INVALID\n"

print(ans)
